Monika After Story/game/script-holidays.rpy

```diff
@@ -4845,3 +4752,1114 @@ label greeting_gone_over_f14_normal_plus:
+
+            "_mas_bday_opened_game": "922.actions.opened_game",
+            "_mas_bday_no_time_spent": "922.actions.no_time_spent",
+            "_mas_bday_no_recognize": "922.actions.no_recognize",
+            "_mas_bday_said_happybday": "922.actions.said_happybday"
+        },
+        exit_pp=store.mas_history._bday_exit_pp
+    ))
+
+############### [HOL060]: METHODS
+init -1 python:
+    def mas_isMonikaBirthday():
+        return datetime.date.today() == mas_monika_birthday
+
+
+    def mas_getNextMonikaBirthday():
+        today = datetime.date.today()
+        if mas_monika_birthday < today:
+            return datetime.date(
+                today.year + 1,
+                mas_monika_birthday.month,
+                mas_monika_birthday.day
+            )
+        return mas_monika_birthday
+
+
+    def mas_recognizedBday(_date=None):
+        """
+        Checks if the user recognized monika's birthday at all.
+
+        RETURNS:
+            True if the user recoginzed monika's birthday, False otherwise
+        """
+        if _date is None:
+            _date = mas_monika_birthday
+
+        return (
+            mas_generateGiftsReport(_date)[0] > 0
+            or persistent._mas_bday_date_affection_gained > 0
+            or persistent._mas_bday_sbp_reacted
+            or persistent._mas_bday_said_happybday
+        )
+
+    def mas_surpriseBdayShowVisuals(cake=False):
+        """
+        Shows bday surprise party visuals
+        """
+        if cake:
+            renpy.show("mas_bday_cake_monika", zorder=store.MAS_MONIKA_Z+1)
+        renpy.show("mas_bday_banners", zorder=7)
+        renpy.show("mas_bday_balloons", zorder=8)
+
+
+    def mas_surpriseBdayHideVisuals():
+        """
+        Hides all visuals for surprise party
+        """
+        renpy.hide("mas_bday_banners")
+        renpy.hide("mas_bday_balloons")
+
+    def mas_confirmedParty():
+        """
+        Checks if the player has confirmed the party
+        """
+        #Must be within a week of the party (including party day)
+        if mas_monika_birthday - datetime.timedelta(days=7) <= today <= mas_monika_birthday:
+            #If this is confirmed already, then we just return true, since confirmed
+            if persistent._mas_bday_confirmed_party:
+                #We should also handle if the player confirmed the party pre-note
+                if persistent._mas_bday_hint_filename:
+                        store.mas_docking_station.destroyPackage(persistent._mas_bday_hint_filename)
+                return True
+
+            #Otherwise, we need to check if the file exists (we're going to make this as foolproof as possible)
+            #Step 1, get the characters folder contents
+            char_dir_files = store.mas_docking_station.getPackageList()
+
+            #Step 2, We need to remove the extensions
+            for file in char_dir_files:
+                temp_filename = file[:file.index('.')] if "." in file else file
```

Comment:
`partition` and related functions return tuples, first element is the string you called it on even if the separator isn't found so its better to use.
```suggestion
                temp_filename = file.rpartition(".")[0]
```